pkg/resources/kafka/kafka.go

```diff
@@ -283,11 +284,21 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 		if err = r.updateStatusWithDockerImageAndVersion(broker.Id, brokerConfig, log); err != nil {
 			return err
 		}
-		if err = r.reconcilePerBrokerDynamicConfig(broker.Id, brokerConfig, configMap, log); err != nil {
-			return err
+		// If dynamic configs can not be set then let the loop continue to the next broker,
+		// after the loop we return error. This solve that case when other brokers could get healthy,
+		// but the loop exits too soon because dynamic configs can not be set.
+		err = r.reconcilePerBrokerDynamicConfig(broker.Id, brokerConfig, configMap, log)
+		if err != nil {
+			log.Error(err, "setting dynamic configs has failed", "brokerID", broker.Id)
+			isPerBrokerDynamicConfigFailed = true
 		}
 	}
 
+	if isPerBrokerDynamicConfigFailed {
+		// re-reconcile to retry setting the dynamic configs
+		return errors.New("setting dynamic configs for some brokers has failed.")
```

Comment:
Add context to the error (e.g. see https://github.com/banzaicloud/koperator/blob/master/pkg/resources/kafka/kafka.go#L304-L307)